[13013] Convert new appt form class components to function components that use hooks

LanceGundersen opened the pull request:
## Description
We have a lot of class components (mostly page components) that could be converted to function components and use hooks for whatever stateful behavior is needed.

- [x] ClinicChoicePage
- [x] CommunityCarePreferencesPage
- [x] ContactInfoPage
- [x] PreferredDatePage
- [x] ReasonForAppointmentPage
- [x] ConfirmationPage
- [x] ReviewPage

## Testing done
Local and Unit

## Screenshots
n/a

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

LanceGundersen committed:
Convert CommunityCarePreferencesPage to function component

LanceGundersen committed:
Convert ContactInfoPage to function component

LanceGundersen committed:
Convert PreferredDatePage to function component

LanceGundersen committed:
Convert ReasonForAppointmentPage to function component WIP

LanceGundersen committed:
Convert ClinicChoicePage to function component

LanceGundersen committed:
Convert the ReasonForAppointmentPage to function component

LanceGundersen committed:
Add pageKey and change pageUISchema

LanceGundersen committed:
Fix test chars assertion

LanceGundersen committed:
Convert the ConfirmationPage to function component

LanceGundersen committed:
Convert the ReviewPage to function component

LanceGundersen committed:
Fix the update change

LanceGundersen committed:
Remove typeOfCare prop type

LanceGundersen committed:
Fix action import

LanceGundersen committed:
Fix history

jbalboni commented on src/applications/vaos/new-appointment/components/ClinicChoicePage.jsx:
```diff
@@ -45,142 +40,111 @@ const initialSchema = {
+  useEffect(() => {
+    document.title = `${getPageTitle(schema, typeOfCare)} | Veterans Affairs`;
+    openClinicPage(pageKey, uiSchema, initialSchema);
     scrollAndFocus();
```
It looks like the previous intent of this page was to again scrollAndFocus when we display an h1, which happens after loading has finished.

I think I would create another useEffect with `schema && facilityDetailsStatus !== FETCH_STATUS.loading` as a dependency

LanceGundersen replied:
Done

jbalboni commented on src/applications/vaos/new-appointment/components/PreferredDatePage.jsx:
```diff
@@ -48,64 +44,61 @@ const uiSchema = {
+  openFormPage: actions.openFormPage,
+  routeToPreviousAppointmentPage: actions.routeToPreviousAppointmentPage,
+  routeToNextAppointmentPage: actions.routeToNextAppointmentPage,
+  submitExpressCareRequest: actions.submitExpressCareRequest,
```
Copy and paste error?

LanceGundersen replied:
Nice catch.

LanceGundersen commented on src/applications/vaos/new-appointment/components/ConfirmationPage/index.jsx:
```diff
@@ -25,97 +22,85 @@ import {
-      this.props.fetchFacilityDetails(data.vaFacility);
-    }
-    scrollAndFocus();
+  if (
```
Done

LanceGundersen replied:
@jbalboni I am getting an error when I do this for references but I don't see where the property references is.
`TypeError: Cannot read property 'references' of undefined
Occurred while linting /Users/lancegundersen/dev/vets-website/src/applications/vaos/new-appointment/components/ConfirmationPage/index.jsx:40`

LanceGundersen commented on src/applications/vaos/new-appointment/components/ConfirmationPage/index.jsx:
```diff
@@ -25,97 +22,85 @@ import {
+  }, []);
 
-    this.pageTitle = pageTitle;
+  if (history && !data?.typeOfCareId) {
```
Done

LanceGundersen committed:
Feedback fixes

- ConfirmationPage move if statements to useEffect
- ClinicChoicePage add useEffect for loaded
- PreferredDatePage remove copy paste error

jbalboni commented on src/applications/vaos/new-appointment/components/ClinicChoicePage.jsx:
```diff
@@ -45,142 +40,118 @@ const initialSchema = {
+    data.clinicId === 'NONE' && !canMakeRequests;
+  useEffect(() => {
+    openClinicPage(pageKey, uiSchema, initialSchema);
     scrollAndFocus();
```
I think you can remove `scrollAndFocus` here, since it'll get called in the other `useEffect` twice: once on mount, then again when your conditional in the dependency array changes.

LanceGundersen replied:
Good call, I wasn't sure so that answers my question too.

jbalboni reviewed (approved):
One comment, LGTM otherwise

LanceGundersen committed:
Remove unneeded scrollAndFocus
